### PR TITLE
Remove unix constraint from cabal.project

### PR DESCRIPTION
**Pull request by phadej**
Add bunch of installed constraints to cabal.project.local.travis

---
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [ ] Any changes that could be relevant to users have been recorded in the changelog.
* [ ] The documentation has been updated, if necessary.
* [ ] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

**Comment by phadej** [comment]
Will fail with

```
[ 10 of 220] Compiling Distribution.Compat.Directory ( Distribution/Compat/Directory.hs, /home/ogre/Documents/shared-haskell/cabal/dist-newstyle/build/x86_64-linux/ghc-7.8.4/Cabal-2.4.1.0/build/Distribution/Compat/Directory.o ) [System.Directory changed]

Distribution/Compat/Directory.hs:41:27:
    Not in scope: ‘<$>’
    Perhaps you meant one of these:
      ‘</>’ (imported from System.FilePath),
      ‘<.>’ (imported from System.FilePath)

Distribution/Compat/Directory.hs:41:55:
    Not in scope: ‘<*>’
    Perhaps you meant one of these:
      ‘</>’ (imported from System.FilePath),
      ‘<.>’ (imported from System.FilePath)
```

**Comment by phadej** [comment]
https://github.com/haskell-infra/hackage-trustees/issues/196

File: cabal.project [hunk 1]
```diff
@@ -1,5 +1,4 @@
 packages: Cabal/ cabal-testsuite/ cabal-install/ solver-benchmarks/ pretty-show-1.6.16/
-constraints: unix >= 2.7.1.0
```
Comment (phadej) [inline cabal.project]: If this breaks OSX build, then let's have an own `cabal.project` for `OSX`.

Comment (phadej) [reply cabal.project]: ... or

if os(osx)
  build-depends: unix >= 2.7.1.0

**Comment by phadej** [comment]
```
Test/Cabal/Server.hs:257:45:
    Couldn't match type `[Char]' with `(ProcessHandle__, [Char])'
```

argh...

**Comment by phadej** [comment]
ping @23Skidoo I *think* `Cabal` should be fine with GHC-7.8 and 7.6 now.

**Comment by 23Skidoo** [comment]
Great, thanks. I'll wait a week or so before making a bugfix lib:Cabal release.

**Pull request merged**